src/components/ui/drawers/names/NameEditionDrawer.tsx

```diff
@@ -0,0 +1,129 @@
+import { FormProvider, SubmitHandler, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { toast } from '../../use-toast';
+import { api } from '@/lib/hono';
+import { revalidatePage } from '@/actions/revalidation';
+import { usePathname } from 'next/navigation';
+import { TOAST_TEXTS } from '@/constants/toast-texts';
+import {
+  getNameEditionDrawerSchema,
+  NameEditionDrawerContent,
+  TNameEditionDrawerSchema,
+} from '@/components/ui/drawers/names/NameEditionDrawerContent';
+
+interface INameEditionDrawer {
+  name: string;
+  open: boolean;
+  setOpen: (value: boolean) => void;
+  type: 'sharehouse' | 'category';
+  id?: string;
+}
+
+/**
+ * A drawer component to edit a sharehouse or category name
+ *
+ * @param name - The current name to be edited
+ * @param open - The state of the drawer
+ * @param setOpen - The function to set the state of the drawer
+ * @param type - The type to determine if editing a 'sharehouse' or 'category'
+ *
+ * @example
+ * const [open, setOpen] = useState(false);
+ * // To edit the sharehouse name
+ * <NameEditionDrawer
+ *  name={'sample name'}
+ *  open={open}
+ *  setOpen={setOpen}
+ *  type={'sharehouse'}
+ * />
+ *
+ * // To edit the category name
+ * <NameEditionDrawer
+ *  name={'sample name'}
+ *  open={open}
+ *  setOpen={setOpen}
+ *  type={'category'}
+ * />
+ */
```

Comment:
If possible, could you also add the `id` prop to the @param and @example?

Reply:
@Akikaze1119 Hi. I've just added!